storage: log Epoch lease only when changing hands

tbg opened the pull request:
This should noticeably denoise the logs for clusters that have
lots of ranges. After restarting a node in such a cluster, a
long tail of these messages would crop up as the various queues
acquired lease on otherwise dormant ranges previously lead by
the restarted node.

I think we'll need to do more to reign in these messages in
the future, but this is a good start.

Release note: None

a-robinson reviewed (approved):
LGTM

I imagine no one was relying on being able to get this message for all ranges (regardless of whether a lease is changing hands) when running at high verbosity? I'd assume no one was.

tbg committed:
storage: log Epoch lease only when changing hands

This should noticeably denoise the logs for clusters that have
lots of ranges. After restarting a node in such a cluster, a
long tail of these messages would crop up as the various queues
acquired lease on otherwise dormant ranges previously lead by
the restarted node.

I think we'll need to do more to reign in these messages in
the future, but this is a good start.

Release note: None

tbg reviewed (commented):
I don't think anyone was, but I think you have a point. Changed so that V(1) always logs.

bors r=a-robinson

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29113#-:-LKwh-AS1SMU7yiwK1Sb:b51zhv3)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->
